djstripe/models/billing.py

```diff
@@ -1322,15 +1392,50 @@ class Subscription(StripeModel):
         help_text="The quantity applied to this subscription. This value will be "
         "`null` for multi-plan subscriptions",
     )
-    start = StripeDateTimeField(
+    # TODO: schedule (implement model SubscriptionSchedule, see #899)
+    legacy_start = StripeDateTimeField(
         help_text="Date of the last substantial change to "
         "this subscription. For example, a change to the items array, or a change "
         "of status, will reset this timestamp."
     )
+    start_date = StripeDateTimeField(
+        null=True,
+        blank=True,
+        help_text="Date when the subscription was first created. The date "
+        "might differ from the created date due to backdating.",
+    )
     status = StripeEnumField(
         enum=enums.SubscriptionStatus, help_text="The status of this subscription."
     )
-    tax_percent = StripePercentField(
+
+    @property
+    def billing(self):
+        warnings.warn(
+            "Invoice.billing has been renamed to .collection_method. "
+            "This alias will be removed in djstripe 2.4",
+            DeprecationWarning,
+        )
+        return self.collection_method
+
+    @property
+    def start(self):
+        warnings.warn(
+            "Invoice.start is deprecated and this alias will be removed in "
+            "djstripe 2.4. USe Invoice.start_date instead.",
+            DeprecationWarning,
+        )
+        return self.legacy_start
+
+    @property
+    def tax_percent(self):
+        warnings.warn(
+            "Subscription.tax_percent has been deprecated and will be removed "
+            "in djstripe 2.4",
+            DeprecationWarning,
+        )
+        return self.legacy_tax_percent
+
+    legacy_tax_percent = StripePercentField(
```

Comment:
I think we're previously just put a warning in HISTORY eg "tax_percent has been deprecated by stripe and will be dropped in djstripe 2.4" rather than adding a rename to legacy migration.

I don't object though, just creates a bit more work to manage them.

Note that the reason I did a legacy_status rename in a previous PR which I think may have inspired you was because there was a direct name clash between a field we had defined and a new one added by stripe - I wanted to try and avoid the case that people switched from our home-brew status field to Stripe's without realising.

Reply:
You're right. I've been inspired by the `status` to `legacy_status` change.

What do you suggest, just remove it in the next release without deprecating it previously? I agree on this way too since minor releases are generally allowed to include backwards incompatible changes. I'm just unsure on what is the usual procedure here.

Reply:
I'd suggest adding to the change log that `start` is deprecated and will be removed in 2.4, and add a comment before the field that it will be removed.

But probably don't bother with the rename to `legacy_date`.